src/ppx/scene/scene_material.cpp

```diff
@@ -183,22 +205,25 @@ scene::VertexAttributeFlags MaterialFactory::GetRequiredVertexAttributes(const s
     return attrFlags;
 }
 
-scene::MaterialRef MaterialFactory::CreateMaterial(
+scene::Material* MaterialFactory::CreateMaterial(
```

Comment:
Since we are switching from shader_ptr to raw pointer, its up to the callers to manage its lifetime?

Reply:
The ref is still used in other places.